### DISCUSSION TITLE
Add Params(map[string][]string) and Headers(map[string][]string) methods

### DISCUSSION THREAD

**Discussion by earthboundkid** [earthboundkid/requests#23] [Ideas] [answered]
It would be nice to have a way to bulk set query params and the same for headers.

**Comment by earthboundkid** [comment]
From #76:

> Sometimes you need to add parameters conditionally, and it's annoying to break the flow with an if statement, so having a Params method might be nice.
> 
> Questions:
> 
> * Should it be `Params(map[string]string)`, `Params(map[string][]string)`, `Params([]string)` (panic if len(strings) isn't even), or `Params(...string)`?
> * Does it make sense to have `Headers`? Seems less useful.

**Reply by earthboundkid** [reply]
If it's map[string][]string, then you can potentially pass in url.Values or http.Header objects. 

Could go crazy and do `Params(any)` and then figure it out from context. 😆 🤔

**Reply by earthboundkid** [reply]
Instead of `Params(any)`, which is crazy, there could be `requests.Map(any)` which takes whatever and inflates it into a map[string][]string.

**Reply by tmm1** [reply]
:+1: ran into this myself. An easy way to pass through `url.Values` would be appreciated.

**Comment by earthboundkid** [comment, marked as answer]
See #82.
